src/OpenTelemetry/BatchExportProcessor.cs

```diff
@@ -217,7 +226,17 @@ private void ExporterProc()
 
                 if (this.circularBuffer.Count > 0)
                 {
-                    this.exporter.Export(new Batch<T>(this.circularBuffer, this.maxExportBatchSize));
+                    try
+                    {
+                        this.exporter.Export(new Batch<T>(this.circularBuffer, this.maxExportBatchSize));
+                    }
+                    catch (Exception ex)
+                    {
+                        lock (this.exporterProcException)
```

Comment:
instead of taking lock, why not just log to EventSource right here?

Reply:
Yeah that makes sense. I have changed the code to log to EventSource in the catch block.